packages/terra-form-select/src/shared/_MenuUtil.js

```diff
@@ -195,6 +195,9 @@ class MenuUtil {
   static findNext(object, value) {
     const options = MenuUtil.flatten(object, true);
     const index = options.findIndex(({ props }) => MenuUtil.isEqual(props.value, value));
+    if (options.length - 1 === index) {
+      return options[0].props.value;
+    }
     return index === -1 ? null : options[Math.min(index + 1, options.length - 1)].props.value;
```

Comment:
Instead of having two return statements, we should be able to simply change the value of the index variable and then return the options value at that index.  You would no longer need the Math.min function because you are already checking for the boundaries.

Reply:
Removed Math.max function.
If we change the index value to -1 then null value will be return from another return statement